src/__tests__/index.test.ts

```diff
@@ -400,3 +401,54 @@ describe('deleteSubNotification by subNotificationId', () => {
     );
   });
 });
+
+describe('setUserPreferences without subNotificationId', () => {
+  const retractEndPointRegex = /.*\/user_preferences\/.*/;
+  const clientId = 'testClientId';
+  const clientSecret = 'testClientSecret';
+  const userId = 'testUserId';
+  const userPreferences: SetUserPreferences[] = [
+    {
+      notificationId: 'notificationId',
+      channel: Channels.EMAIL,
+      state: true
+    }
+  ];
+  test('makes API calls to the correct end-point', async () => {
+    axiosMock.onPost(retractEndPointRegex).reply(200);
+    await notificationapi.init(clientId, clientSecret);
+    await notificationapi.setUserPreferences(userId, userPreferences);
+    expect(axiosMock.history.post).toHaveLength(1);
+    expect(axiosMock.history.post[0].url).toEqual(
+      `https://api.notificationapi.com/${clientId}/user_preferences/${userId}`
+    );
+    expect(axiosMock.history.post[0].data).toEqual(
```

Comment:
When asserting completely different things, separate test cases.

Check the retract tests